src/Datasource/EntityTrait.php

```diff
@@ -360,6 +360,20 @@ public function unsetProperty($property)
     }
 
     /**
+     * Get the list of all properties.
+     *
+     * The list of all properties is all standard properties
+     * plus virtual properties plus hidden properties.
+     *
+     * @return array A list of all entity properties
+     */
+    public function allProperties()
```

Comment:
Shouldn't this method return all property values instead of just properties names?  In it's current form the patch still adds optional param to `toArray()` which is a API breaking change.